pkgs/tools/misc/dua/default.nix

```
@@ -7,21 +7,23 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "dua";
-  version = "2.29.0";
+  version = "2.29.1";
 
   src = fetchFromGitHub {
     owner = "Byron";
     repo = "dua-cli";
     rev = "v${version}";
-    hash = "sha256-rO9k1/HOwVJF/QCT2sZy4L0Mv26CiUj9Zafliffj68A=";
+    hash = "sha256-YOXK6u9KF2FnICMFEj5IEZG2OQ6PaKQsShp7/gbSK4Y=";
     # Remove unicode file names which leads to different checksums on HFS+
     # vs. other filesystems because of unicode normalisation.
     postFetch = ''
       rm -r $out/tests/fixtures
     '';
   };
 
-  cargoHash = "sha256-qn1QDiYHcygomOFwFEy00wsMykrQ9/84Ed4nAUTlA1k=";
+  cargoPatches = [ ./add-cargo-lock.patch ];
+
+  cargoHash = "sha256-Km8I5t6xDaCYn4Z8v7LLDtX4sk89iqnliECPiV7Ed5o=";
```

The Nixpkgs manual suggests doing it like this, which avoids generating a patch file and updating the hash every time:
```suggestion
  cargoLock.lockFile = ./Cargo.lock;

  postPatch = ''
    ln -s ${./Cargo.lock} Cargo.lock
  '';
```